azext_edge/edge/util/file_operations.py

```diff
@@ -0,0 +1,135 @@
+
+
+def convert_file_content_to_json(file_path: str) -> List[Dict[str, Any]]:
+    extension = file_path.split(".")[-1]
+    invalid_extension = extension not in ["json", "yaml", "yml", "csv"]
+    content = read_file_content(file_path)
+    result: Optional[List[Dict[str, Any]]] = None
+    if invalid_extension or extension == "json":
+        result = try_loading_as(
+            loader=json.loads,
+            content=content,
+            error_type=json.JSONDecodeError,
+            raise_error=not invalid_extension
+        )
+    if (not result and invalid_extension) or extension in ["yaml", "yml"]:
+        result = try_loading_as(
+            loader=yaml.safe_load,
+            content=content,
+            error_type=yaml.YAMLError,
+            raise_error=not invalid_extension
+        )
+    if (not result and invalid_extension) or extension == "csv":
+        result = try_loading_as(
+            loader=csv.DictReader,
+            content=content.splitlines(),
+            error_type=csv.Error,
+            raise_error=not invalid_extension
+        )
+    if result is not None:
+        return result
+    raise FileOperationError(f"File contents for {file_path} cannot be read.")
+
+
+# TODO: unit test
+def dump_content_to_file(
+    content: List[dict],
+    file_name: str,
+    extension: str,
+    fieldnames: Optional[List[str]] = None,
+    output_dir: Optional[str] = None,
+    replace: bool = False
+) -> PurePath:
+    output_dir = normalize_dir(output_dir)
+    file_path = os.path.join(output_dir, f"{file_name}.{extension}")
+    if os.path.exists(file_path):
+        if not replace:
+            raise FileExistsError(f"File {file_path} already exists. Please choose another file name or add replace.")
+        logger.warning(f"The file {file_path} will be overwritten.")
+    if extension.endswith("csv"):
+        with open(file_path, "w", newline="", encoding="utf-8") as f:
+            if not fieldnames:
+                fieldnames = content[0].keys()
+            writer = csv.DictWriter(f, fieldnames=fieldnames)
+            writer.writeheader()
+            writer.writerows(content)
+        return file_path
+
+    # These let you dump to a string before writing to file
+    if extension == "json":
+        content = json.dumps(content, indent=2)
+    elif extension in ["yaml", "yml"]:
+        content = yaml.dump(content)
+    with open(file_path, "w", encoding="utf-8") as f:
+        f.write(content)
+
+    return file_path
+
+
+def normalize_dir(dir_path: Optional[str] = None) -> PurePath:
+    if not dir_path:
+        dir_path = "."
+    if "~" in dir_path:
+        dir_path = os.path.expanduser(dir_path)
+    dir_path = os.path.abspath(dir_path)
+    dir_pure_path = PurePath(dir_path)
+    if not os.path.exists(str(dir_pure_path)):
+        os.makedirs(dir_pure_path, exist_ok=True)
+
+    return dir_pure_path
+
+
+def read_file_content(file_path: str, read_as_binary: bool = False) -> Union[bytes, str]:
```

Comment:
The generic functions in this module are `read_file_content` and `normalize_dir`.

What is currently called `convert_file_content_to_json` and its usage of `try_loading_as` is too specific of an implementation to be generally useful.